src/compiler/scanner.ts

```diff
@@ -626,13 +626,13 @@ namespace ts {
      * @returns If "reduce" is true, the accumulated value. If "reduce" is false, the first truthy
      *      return value of the callback.
      */
-    function iterateCommentRanges<T, U>(reduce: boolean, text: string, pos: number, trailing: boolean, cb: (pos: number, end: number, kind: CommentKind, hasTrailingNewLine: boolean, state: T, memo: U) => U, state: T, initial?: U): U {
+    function iterateCommentRanges<T, U>(reduce: boolean, text: string, pos: number, trailing: boolean, cb: (pos: number, end: number, kind: CommentKind, hasTrailingNewLine: boolean, state: T, memo: U) => U, state: T, initial?: U, inline?: boolean): U {
         let pendingPos: number;
         let pendingEnd: number;
         let pendingKind: CommentKind;
         let pendingHasTrailingNewLine: boolean;
         let hasPendingCommentRange = false;
-        let collecting = trailing || pos === 0;
+        let collecting = inline || trailing || pos === 0;

```

Comment:
Don't change this. I assure you, though the terminology is confusing, it is at least consistent. What you want to change instead is the position you begin scanning from, waaaaay up the callstack closer to `shouldStripInternal`. Rather than looking for the node's leading or trailing comments, you want the trailing comments of the first non-trivia position _before_ the node. For example, in:
```ts
(/* whatever*/ Foo);
```
The `/*whatever*/` comment is neither a leading nor trailing comment of the `Foo` identifier. Nor is is a leading or trailing comment of the parenthesized expression/argument list. What it _is_ is a trailing comment of the `(` - the OpenParen token. This token _doesn't actually exist in our AST_, so recovering comments on it can be difficult. But in this case it's not hard - you just want the trailing comments of the argument list start for the first parameter, or the trailing comments of the end of the previous parameter + skip trivia length + 1 if it's not the first param. Our emitter has a bunch of code that does similar stuff to preserve comments on these tokens - checkout the `emitTrailingCommentsOfPosition` and `emitLeadingCommentsOfPosition` calls in `emitNodeList` in `emitter.ts`.